EIPS/eip-flash-loans.md

```diff
@@ -0,0 +1,165 @@
+
+The `flashLoan` function MUST execute the equivalent of an `ERC20.transfer` operation before calling `FlashBorrowerLike(receiver).onFlashLoan(...)`.
+
+The lender contract MAY `mint` the tokens lended, instead of executing a `transfer` of tokens it holds.
+
+The `flashLoan` function MUST verify that the tokens lended were returned, and MUST NOT take them from the `receiver`.
+
+The `receiver` MUST take an action to return `amount + fee` tokens and allow the transaction to resolve.
+
+If the `flashLoan` used tokens generated by a `mint`, they SHOULD be the target of a `burn` before the end of the transaction.
+
+If a fee is charged, the contract implementing `flashLoan` MAY use it in any desired way (e.g. the fee can be burned or transferred to any other party).
+
+## Rationale
+
+The interfaces described in this ERC have been chosen as to cover the known flash lending use cases, while allowing for safe and gas efficient implementations.
+
+`flashLoan(address receiver, uint256 amount, bytes calldata data)`
+
+`flashLoan` has been chosen as descriptive enough, unlikely to clash with other functions in the lender, and including both the use cases in which the tokens lended are held or minted by the lender.
+
+`receiver` is taken as a parameter to allow `flashLoan` to be called by EOAs, as opposed to the pattern in which `onFlashLoan` is called on `msg.sender`. This allows the lender to inform the `receiver` which address called `flashLoan`. This particular setup allows the `receiver` to implement an account based platform.
+
+A `bytes calldata data` parameter is included for the caller to pass arbitrary information to the `receiver`, without impacting the utility of the `flashMint` standard.
+
+`onFlashLoan(msg.sender, amount, fee, data)`
+
+`onFlashLoan` has been chosen as descriptive enough, unlikely to clash with other functions in the `receiver`, and following the `onAction` naming pattern used as well in EIP-667.
+
+A `user` will often be required in the `onFlashLoan` function, which the lender knows as `msg.sender`. An alternative implementation which would embed the `user` in the `data` parameter by the caller would require an additional mechanism for the receiver to verify its accuracy, and is not advisable.
+
+The `amount` will be required in the `onFlashLoan` function, which the lender took as a parameter. An alternative implementation which would embed the `amount` in the `data` parameter by the caller would require an additional mechanism for the receiver to verify its accuracy, and is not advisable.
+
+A `fee` will often be calculated in the `flashMint` function, which the `receiver` must be aware of for repayment. Passing the `fee` as a parameter instead of appended to `data` is simple and effective.
+
+
+## Backwards Compatibility
+
+No backwards compatibility issues identified.
+
+## Test Cases
+
+[WETH10](https://github.com/WETH10/WETH10/pull/81), [MakerDAO](https://github.com/hexonaut/dss-flash/pull/18).
+
+## Implementation
```

Comment:
It's preferred for a reference implementation to be included inline.

Reply:
Included now. What should I do with regards to tests? Would a link be enough?

Reply:
A link is okay. Can you update the link to where the test cases are, rather than a PR please?